File: room-map-receiver/db/model/room.rb
# frozen_string_literal: true

class Room < ActiveRecord::Base
end

class CreateRooms < ActiveRecord::Migration[6.0]
  def change
    create_table :rooms do |t|
      t.string :title
      t.string :origin
      t.string :url
      t.string :code
      t.date :publish_time

      t.float :price
      t.float :price_per_sqm
      t.string :price_type
      t.string :price_rent
      t.string :price_deposit
      t.string :price_service
      t.string :price_agent

      t.string :position_district
      t.string :position_region
      t.string :position_community
      t.decimal :position_longitude, precision: 10, scale: 6
      t.decimal :position_latitude, precision: 10, scale: 6

      t.string :lease_type

      t.string :house_layout
      t.integer :house_area
      t.string :house_face
      t.string :house_floor
      t.string :house_lift
      t.string :house_water
      t.string :house_electric
      t.string :house_gas
      t.string :house_heating

      t.timestamps
    end
    add_index :rooms, :code, unique: true
  end
end

CreateRooms.new.change unless ActiveRecord::Base.connection.table_exists? 'rooms'

File: room-map-receiver/src/entity/Room.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity()
export class Room {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  title: string;

  @Column()
  origin: string;

  @Column()
  url: string;

  @Column()
  code: string;

  @Column({ type: 'timestamp' })
  publish_time: Date;

  // ===================================

  @Column({ type: 'float' })
  price: number;

  @Column({ type: 'float' })
  price_per_sqm: number;

  @Column({ type: 'simple-array' })
  price_type: string[];

  @Column({ type: 'simple-array' })
  price_rent: string[];

  @Column({ type: 'simple-array' })
  price_deposit: string[];

  @Column({ type: 'simple-array' })
  price_service: string[];

  @Column({ type: 'simple-array' })
  price_agent: string[];

  // ===================================

  @Column()
  position_district: string;

  @Column()
  position_region: string;

  @Column()
  position_community: string;

  @Column({ type: 'decimal' })
  position_longitude: number;

  @Column({ type: 'decimal' })
  position_latitude: number;

  // ===================================

  @Column()
  lease_type: string;

  // ===================================

  @Column()
  house_layout: string;

  @Column({ type: 'int' })
  house_area: number;

  @Column()
  house_face: string;

  @Column()
  house_floor: string;

  @Column()
  house_lift: string;

  @Column()
  house_water: string;

  @Column()
  house_electric: string;

  @Column()
  house_gas: string;

  @Column()
  house_heating: string;

  // ===================================

  @Column({ type: 'timestamp' })
  create_time: Date;

  @Column({ type: 'timestamp' })
  update_time: Date;
}
